File: Dataset/Leetcode/test/22/702.py
 class Solution:
    def XXX(self, n: int) -> List[str]:
        ans = [(0,0,'')]
        while True:
            t = []
            for a,b,s in ans:
                if a>0:t.append((a-1,b,s+')'))
                if b<n:t.append((a+1,b+1,s+'('))
            if t:ans = t
            else:break
        return list(map(lambda x:x[2],ans))

File: Dataset/Leetcode/train/11/702.java
 class Solution {
    public int XXX(int[] height) {
        int i = 0;
        int k = 1;
        int max = 0;
        while(i < height.length) {
            if(k > height.length -1) {
                i++;
                k = i + 1;
                continue;
            }
            int width = k-i;
            int min = height[i] < height[k] ? height[i] : height[k];
            if (width * min > max) {
                max = width * min;
            }
            k++; 
        }
        return max;
    }
}

